packages/build/tests/monitor/snapshots/tests.js.md

```diff
@@ -1782,6 +1782,86 @@ Generated by [AVA](https://avajs.dev).
+      Netlify Build                                                 ␊
+    ────────────────────────────────────────────────────────────────␊
+    ␊
+    > Version␊
+      @netlify/build 1.0.0␊
+    ␊
+    > Flags␊
+      debug: true␊
+      repositoryRoot: packages/build/tests/monitor/fixtures/function_bundling_error␊
+      testOpts:␊
+        errorMonitor: true␊
+        pluginsListUrl: test␊
+        silentLingeringProcesses: true␊
+    ␊
+    > Current directory␊
+      packages/build/tests/monitor/fixtures/function_bundling_error␊
+    ␊
+    > Config file␊
+      No config file was defined: using default values.␊
+    ␊
+    > Resolved config␊
+      build:␊
+        publish: packages/build/tests/monitor/fixtures/function_bundling_error␊
+        publishOrigin: default␊
+      functionsDirectory: packages/build/tests/monitor/fixtures/function_bundling_error/netlify/functions␊
+    ␊
+    > Context␊
+      production␊
+    ␊
+    ────────────────────────────────────────────────────────────────␊
+      1. Functions bundling                                         ␊
+    ────────────────────────────────────────────────────────────────␊
+    ␊
+    Packaging Functions from netlify/functions directory:␊
+     - trouble.ts␊
+    ␊
+    ␊
+    ────────────────────────────────────────────────────────────────␊
+      Bundling of function "trouble" failed                         ␊
+    ────────────────────────────────────────────────────────────────␊
+    ␊
+      Error message␊
+      Build failed with 1 error:␊
+      packages/build/tests/monitor/fixtures/function_bundling_error/netlify/functions/trouble.ts ERROR: Cannot assign to "bar" because it is a constant␊
+    ␊
+      Error location␊
+      While bundling function "trouble"␊
+    ␊
+      Resolved config␊
+      build:␊
+        publish: packages/build/tests/monitor/fixtures/function_bundling_error␊
+        publishOrigin: default␊
+      functionsDirectory: packages/build/tests/monitor/fixtures/function_bundling_error/netlify/functions␊
+    ␊
+    Error monitoring payload:␊
+    {␊
+      "errorClass": "functionsBundling",␊
+      "errorMessage": "Build failed with 1 error:/npackages/build/tests/monitor/fixtures/function_bundling_error/netlify/functions/trouble.ts ERROR: Cannot assign to /"bar/" because it is a constant",␊
+      "context": "Bundling of function failed",␊
+      "groupingHash": "Bundling of function failed/nBuild failed with 0 error:/n/file/path ERROR: Cannot assign to /"bar/" because it is a constant",␊
+      "severity": "info",␊
+      "unhandled": false,␊
+      "location": {␊
+        "functionName": "trouble",␊
+        "runtime": "js",␊
+        "bundler": "esbuild"␊
+      },␊
+      "pluginPackageJson": false,␊
+      "BUILD_ID": "0",␊
+      "other": {␊
+        "groupingHash": "Bundling of function failed/nBuild failed with 0 error:/n/file/path ERROR: Cannot assign to /"bar/" because it is a constant"␊
```

Comment:
The has still contains "bar". :/

Reply:
Yep, we'll need to handle that separately. I'm thinking that instead of [completely overriding](https://github.com/netlify/build/blob/main/packages/build/src/error/monitor/report.js#L56-L59) Netlify Build's normalisation rules, the `normalizedMessage` should be used instead of `error.message` but it should still go through the normalisation rules.

So in this case, zip-it-and-ship-it would transform `Cannot assign to "bar"` into `Cannot assign to "variable"` and then Netlify Build would normalise the rest.

(It's also possible that I'm overthinking this.)

Reply:
This seems hard, because it is only one of the many error messages. So have rules in zisi for every single syntax error type? But yeah let's discuss this next week.